src/pytorch_lightning/profilers/hpu.py

```diff
@@ -0,0 +1,151 @@
+
+            row_limit: Limit the number of rows in a table, ``-1`` is a special value that
+                removes the limit completely.
+
+            sort_by_key: Attribute used to sort entries. By default
+                they are printed in the same order as they were registered.
+                Valid keys include: ``cpu_time``, ``cpu_time_total``,
+                ``cpu_memory_usage``, ``self_cpu_memory_usage``, ``count``.
+
+            record_module_names: Whether to add module names while recording autograd operation.
+
+            profiler_kwargs: Keyword arguments for the PyTorch profiler. This depends on your PyTorch version
+
+        Raises:
+            MisconfigurationException:
+                If arg ``sort_by_key`` is not present in ``AVAILABLE_SORT_KEYS``.
+                If arg ``schedule`` is not a ``Callable``.
+                If arg ``schedule`` does not return a ``torch.profiler.ProfilerAction``.
+        """
+        super().__init__(
+            dirpath=dirpath,
+            filename=filename,
+            group_by_input_shapes=group_by_input_shapes,
+            export_to_chrome=export_to_chrome,
+            row_limit=row_limit,
+            sort_by_key=sort_by_key or f"{'hpu' if profiler_kwargs.get('use_hpu', False) else 'cpu'}_time_total",
+            record_module_names=record_module_names,
+            **profiler_kwargs,
+        )
+
+        self.profiler: Optional[_PROFILER] = None
+        self._profiler_kwargs["activities"] = self.profile_hpu_activities(self._profiler_kwargs.get("activities", None))
+
+    def profile_hpu_activities(self, activities) -> List["ProfilerActivity"]:  # type: ignore
+        if not _KINETO_AVAILABLE:
+            return activities
+        activities.append(ProfilerActivity.HPU)
+        return activities
+
+    def stop(self, action_name: str) -> None:
+        if action_name in self._recording_map:
+            self._recording_map[action_name].__exit__(None, None, None)
+            del self._recording_map[action_name]
+
+        if not _KINETO_AVAILABLE or self._emit_nvtx:
+            return
+
+        if self.profiler is not None and any(action_name.endswith(func) for func in self.STEP_FUNCTIONS):
+            if self._schedule is not None:
+                self._schedule.pre_step(action_name)
+
+            # the default schedule requires a minimum of 5 steps to properly work: `wait=1, warmup=1, active=3`.
+            # otherwise, this will raise a `segmentation fault`.
+            if self._should_override_schedule():
+                warning_cache.warn(
+                    "The HPUProfiler default schedule will be overridden as there is not enough "
+                    "steps to properly record traces."
+                )
+                self._schedule = None
+                self.profiler.schedule = torch.profiler.profiler._default_schedule_fn
+
+            def on_trace_ready(profiler: _PROFILER) -> None:
```

Comment:
Does this need to be inside another method? or can be defined as static

Reply:
can't be static, need access to self